docs/modules/operation/pages/provisioning/detectors/BsfDetector.adoc

```diff
@@ -0,0 +1,173 @@
+    <parameter key="fileName" value="/opt/opennms/etc/scripts/MinimalBeanShell.bsh"/>
+    <parameter key="runType" value="eval"/>
+</detector>
+----
+
+.BeanShell example `MinimalBeanShell.bsh` script file
+[source, java]
+----
+File testFile = new File("/tmp/TestFile");
+if (testFile.exists()) {
+  return "OK";
+} else {
+  results.put("reason", "file does not exist");
+  return "NOK";
+}
+----
+
+=== Groovy Example
+
+.Groovy example for `run-type` `eval`
+[source, xml]
+----
+<detector name="MinimalGroovy" class="org.opennms.netmgt.provision.detector.bsf.BSFDetector">
+    <parameter key="bsfEngine" value="org.codehaus.groovy.bsf.GroovyEngine"/>
+    <parameter key="langClass" value="groovy"/>
+    <parameter key="fileName" value="/opt/opennms/etc/scripts/MinimalGroovy.groovy"/>
+    <parameter key="runType" value="eval"/>
+</detector>
+----
+
+.Groovy example `MinimalGroovy.groovy` script file for `run-type` `eval`
+[source, java]
+----
+File testFile = new File("/tmp/TestFile");
+if (testFile.exists()) {
+  return "OK";
+} else {
+  results.put("reason", "file does not exist");
+  return "NOK";
+}
+----
+
+.Groovy example for `run-type` `exec`
+[source, xml]
+----
+<detector name="MinimalGroovy" class="org.opennms.netmgt.provision.detector.bsf.BSFDetector">
+    <parameter key="bsfEngine" value="org.codehaus.groovy.bsf.GroovyEngine"/>
+    <parameter key="langClass" value="groovy"/>
+    <parameter key="fileName" value="/opt/opennms/etc/scripts/MinimalGroovy.groovy"/>
+    <parameter key="runType" value="exec"/>
+</detector>
+
+----
+
+.Groovy example `MinimalGroovy.groovy` script file for `run-type` `exec`
+[source, java]
+----
+def testFile = new File("/tmp/TestFile");
+if (testFile.exists()) {
+  results.put("status", "OK")
+} else {
+  results.put("reason", "file does not exist");
+  results.put("status", "NOK");
+}
+----
+
+=== Example Jython
+
+To use the Jython (Java implementation of Python) language an additional library is required.
+Copy a compatible `jython-x.y.z.jar` into the `opennms/lib` folder and restart {page-component-title}.
+That makes Jython available for the BSF Detector.
+
+.Jython example for `run-type` `exec`
+[source, xml]
+----
+<detector name="MinimalJython" class="org.opennms.netmgt.provision.detector.bsf.BSFDetector">
+    <parameter key="bsfEngine" value="org.apache.bsf.engines.jython.JythonEngine"/>
+    <parameter key="langClass" value="jython"/>
+    <parameter key="fileName" value="/opt/opennms/etc/scripts/MinimalJython.py"/>
+    <parameter key="runType" value="exec"/>
+</detector>
+----
+
+.Jython example `MinimalJython.py` script file for `run-type` `exec`
+[source, python]
+----
+from java.io import File
+
+if (File("/tmp/TestFile").exists()):
+        results.put("status", "OK")
+else:
+        results.put("reason", "file does not exist")
+        results.put("status", "NOK")
+----
+
+NOTE: We have to use `run-type` `exec` here because Jython chokes on the +import+ keyword in `eval` mode.
```

Comment:
```suggestion
NOTE: We must use `run-type` `exec` here because Jython chokes on the +import+ keyword in `eval` mode.
```
Is there a better word than "chokes"? It's fine if it's the most accurate. Just curious.

Reply:
Maybe `disagrees`? :joy:

Reply:
I think chokes is fine. :-)